### PR TITLE
Rollback only if dirty

### PR DESCRIPTION
**Pull request by knownasilya**

### CODE REVIEW

File: addon/mixins/data-route.js [hunk 1]
```diff
@@ -3,10 +3,12 @@ import Ember from 'ember';
 export default Ember.Mixin.create({
   resetController: function() {
     var model = this.get('controller.model');
+    
     if(!model.get('isDeleted')) {
       if (model.get('isNew')) {
         model.deleteRecord();
-      } else {
+      }
```
Comment (rwjblue) [inline addon/mixins/data-route.js]: Why the extra newline?

File: addon/mixins/data-route.js [hunk 2]
```diff
@@ -3,10 +3,12 @@ import Ember from 'ember';
 export default Ember.Mixin.create({
   resetController: function() {
     var model = this.get('controller.model');
+    
     if(!model.get('isDeleted')) {
       if (model.get('isNew')) {
         model.deleteRecord();
-      } else {
+      } 
+      else if (model.get('isDirty')) {
```
Comment (rwjblue) [inline addon/mixins/data-route.js]: Rollback with no dirty state should be a no-op.

**Comment by rwjblue** [comment]
Please add a test that demonstrates what this is fixing (so we can avoid a regression in the future).

File: addon/mixins/data-route.js (hunk 2 above)
Comment (knownasilya) [inline addon/mixins/data-route.js]: Ok, I guess this can be ignored :cactus:

**Pull request closed without merging**